services/braille.c

```diff
@@ -0,0 +1,336 @@
+  // Up
+  { 
+    { { 0x5841, 0x0081 }, { 0x5841, 0x0101 }, { 0x5841, 0x0201 }, { 0x5841, 0x0401 }, { 0x5841, 0x0801 }, { 0x5841, 0x1001 }, { 0x58C1, 0x0001 }, { 0x5941, 0x0001 } },
+    { { 0x5821, 0x0081 }, { 0x5821, 0x0101 }, { 0x5821, 0x0201 }, { 0x5821, 0x0401 }, { 0x5821, 0x0801 }, { 0x5821, 0x1001 }, { 0x58A1, 0x0001 }, { 0x5921, 0x0001 } },
+    { { 0x4611, 0x0081 }, { 0x4611, 0x0101 }, { 0x4611, 0x0201 }, { 0x4611, 0x0401 }, { 0x4611, 0x0801 }, { 0x4611, 0x1001 }, { 0x4691, 0x0001 }, { 0x4711, 0x0001 } }, 
+    { { 0x4609, 0x0081 }, { 0x4609, 0x0101 }, { 0x4609, 0x0201 }, { 0x4609, 0x0401 }, { 0x4609, 0x0801 }, { 0x4609, 0x1001 }, { 0x4689, 0x0001 }, { 0x4709, 0x0001 } }
+  },
+  // Down
+  { 
+    { { 0x5839, 0x0083 }, { 0x5839, 0x0105 }, { 0x5839, 0x0209 }, { 0x5839, 0x0411 }, { 0x5839, 0x0821 }, { 0x5839, 0x1041 }, { 0x5883, 0x0001 }, { 0x5905, 0x0001 } },
+    { { 0x5859, 0x0083 }, { 0x5859, 0x0105 }, { 0x5859, 0x0209 }, { 0x5859, 0x0411 }, { 0x5859, 0x0821 }, { 0x5859, 0x1041 }, { 0x5883, 0x0001 }, { 0x5905, 0x0001 } },
+    { { 0x4669, 0x0083 }, { 0x4669, 0x0105 }, { 0x4669, 0x0209 }, { 0x4669, 0x0411 }, { 0x4669, 0x0821 }, { 0x4669, 0x1041 }, { 0x4683, 0x0001 }, { 0x4705, 0x0001 } },
+    { { 0x4671, 0x0083 }, { 0x4671, 0x0105 }, { 0x4671, 0x0209 }, { 0x4671, 0x0411 }, { 0x4671, 0x0821 }, { 0x4671, 0x1041 }, { 0x4683, 0x0001 }, { 0x4705, 0x0001 } }
+  }
+};
+
+const uint16_t BrDotTbl_forced[ MAXBRPHASE ][ MAXBRROW ][ MAXBRCOL ][ MAXBRBANK ] =
+ {
+  // Braittle module table ( original XY, ghost may occur )
+  // Up
+  {
+  { { 0x5041, 0x0081 }, { 0x5041, 0x0101 }, { 0x5041, 0x0201 }, { 0x5041, 0x0401 }, { 0x5041, 0x0801 }, { 0x5041, 0x1001 }, { 0x50C1, 0x0001 }, { 0x5141, 0x0001 } },
+  { { 0x4821, 0x0081 }, { 0x4821, 0x0101 }, { 0x4821, 0x0201 }, { 0x4821, 0x0401 }, { 0x4821, 0x0801 }, { 0x4821, 0x1001 }, { 0x48A1, 0x0001 }, { 0x4921, 0x0001 } },
+  { { 0x4411, 0x0081 }, { 0x4411, 0x0101 }, { 0x4411, 0x0201 }, { 0x4411, 0x0401 }, { 0x4411, 0x0801 }, { 0x4411, 0x1001 }, { 0x4491, 0x0001 }, { 0x4511, 0x0001 } },
+  { { 0x4209, 0x0081 }, { 0x4209, 0x0101 }, { 0x4209, 0x0201 }, { 0x4209, 0x0401 }, { 0x4209, 0x0801 }, { 0x4209, 0x1001 }, { 0x4289, 0x0001 }, { 0x4309, 0x0001 } }
+  },
+  // Down
+  {
+  { { 0x5001, 0x0083 }, { 0x5001, 0x0105 }, { 0x5001, 0x0209 }, { 0x5001, 0x0411 }, { 0x5001, 0x0821 }, { 0x5001, 0x1041 }, { 0x5083, 0x0001 }, { 0x5105, 0x0001 } },
+  { { 0x4801, 0x0083 }, { 0x4801, 0x0105 }, { 0x4801, 0x0209 }, { 0x4801, 0x0411 }, { 0x4801, 0x0821 }, { 0x4801, 0x1041 }, { 0x4883, 0x0001 }, { 0x4905, 0x0001 } },
+  { { 0x4401, 0x0083 }, { 0x4401, 0x0105 }, { 0x4401, 0x0209 }, { 0x4401, 0x0411 }, { 0x4401, 0x0821 }, { 0x4401, 0x1041 }, { 0x4483, 0x0001 }, { 0x4505, 0x0001 } },
+  { { 0x4201, 0x0083 }, { 0x4201, 0x0105 }, { 0x4201, 0x0209 }, { 0x4201, 0x0411 }, { 0x4201, 0x0821 }, { 0x4201, 0x1041 }, { 0x4283, 0x0001 }, { 0x4305, 0x0001 } }
+  }
+};
+
+
+void delay(uint32_t t) {
```

Comment:
static? it may conflict with something in future

Reply:
Thanks, good catch!